sympy/matrices/matrices.py

```diff
@@ -3983,6 +3983,8 @@ def row_join(self, rhs):
         row
         col_join
         """
+        if not self:
+            return rhs
```

Comment:
here and below, this should be `return self.func(rhs)` (or similar) so the same type of matrix represented by self is returned. But I think the other duplicate PR will be doing this, so let's see.